### DISCUSSION TITLE
Argo server & controller readiness/liveness check

### DISCUSSION THREAD

**Discussion by decipher27** [argoproj/argo-workflows#7066] [Q&A]
Could you let me know why we don't have "Readiness probe" in  argo controller? and  "liveness" in Argo server?

**Comment by tooptoop4** [comment]
did u solve? @decipher27

**Reply by dixneuf19** [reply]
Hi, same question here . There is no official health endpoint for argo workflow server, and in the official Helm chart there is only a readiness probe, but no liveness ([source](https://github.com/argoproj/argo-helm/blob/main/charts/argo-workflows/templates/server/server-deployment.yaml). 

First it is a _good practice_ to have both in a Kubernetes app, but more importantly I already had an issue regarding a _stuck_ workflow server, which was fixed with a simple reboot, something a liveness probe would have catch.

Is there any reason to not implement this kind of feature ? Argocd-server does implement this "health" route, and use it in its Helm chart.
If that's not a priority, can we simply use the root `/` as a custom liveness probe (this is what the chart already does for the readiness) ?
